module_info_pull_request/models/module_informations.py

```diff
@@ -0,0 +1,25 @@
+from odoo import api, fields, models
+
+
+class ModuleInformation(models.Model):
+    _inherit = "module.information"
+
+    pr_ids = fields.Many2many("pull.request", string="Active Pull Request")
+    pr_ids_nbr = fields.Integer(compute="_compute_pr_ids_nbr", string="# of Modules")
+
+    @api.depends("pr_ids")
+    def _compute_pr_ids_nbr(self):
+        for record in self:
+            record.pr_ids_nbr = len(record.pr_ids)
+
+    def action_view_pr(self):
+        self.ensure_one()
+        prs = self.mapped("pr_ids")
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "module_info_pull_request.pull_request_action"
+        )
+        if len(prs) > 0:
+            action["domain"] = [("id", "in", prs.ids)]
```

Comment:
passing all ids is not a good practice (the list can be very long)

Moreover it can be done without any python code.
Add a filter in view of the pull.request on repo_id and activate it with the context

```
<button name="%(pull_request_action)d" type="action" context="{'search_default_repo_id': active_id}"/>
```

example : https://github.com/odoo/odoo/blob/264af51db8a11177e2792c1426640f5ae31332f9/addons/maintenance/views/maintenance_views.xml#L907

Reply:
Ok, it's prettier but how can I pass the PR ids to display on pr view?

Reply:
It's work, when I use a search view.